service/src/main/java/com/ericsson/eiffel/remrem/generate/controller/RemremGenerateController.java

```diff
@@ -36,16 +39,30 @@
     * 
     */
     @RequestMapping(value = "/{mp}", method = RequestMethod.POST)
-    public JsonElement generate(@PathVariable String mp, @RequestParam("msgType") String msgType,
+    public ResponseEntity<?> generate(@PathVariable String mp, @RequestParam("msgType") String msgType,
             @RequestBody JsonObject bodyJson) {
         MsgService msgService = getMessageService(mp);
-        if (msgService != null) {
-            return parser.parse(msgService.generateMsg(msgType, bodyJson));
-        } else {
-            return null;
+        String response= "";
+        try{
+            if (msgService != null) {
+            	response = msgService.generateMsg(msgType, bodyJson);
+                if(!response.contains("message")) {
```

Comment:
Even valid Eiffel event may have a message in its response, Better to make  it as a json element and check the field in the json rahter than contains.